lib/CMakeLists.txt

```diff
@@ -1,3 +1,9 @@
+# TODO: Fix the fine grained tabegen dependencies
```

Comment:
Is this a TODO that someone else could read and act upon? Personally I'd prefer it to be tracked with a https://bugs.swift.org bug, and for it to have a clearer explanation of what needs to be changed and why.

My understanding at the moment is that, ideally, LLVM and Clang would export each of their tablegen targets as a separate target, so that subprojects can specify a dependency upon just, for example, the `clang_tablegen` target `ClangDeclNodes`.

Since LLVM and Clang *don't* export every tablegen target separately, Swift currently relies upon the blanket `clang-tablegen-targets` target, which includes every since thing that Clang tablegens. Is that understanding accurate, or am I misreading?

If I am understanding correctly, is the solution to define targets such as `ClangDeclNodes` in `ClangConfig.cmake.in`, like you did for `clang-tablegen-targets` in https://github.com/apple/swift-clang/commit/a51fd2c5b7e60adab864eb9053aca0f2bfcf6514?